game/player.asm

```diff
@@ -0,0 +1,145 @@
+.section "Player" BANK 0 SLOT "ROM"
+
+nop
+
+.struct Player
+    oam_manager_ptr instanceof OAMManager
+    input instanceof Input
+    oam_obj_ptr dw ; Pointer to the requested OAM object
+    ; enabled db
+.endst
+
+.enum $0000
+    player instanceof Player
+.ende
+
+Player_Init:
+    pha
+    phy
+    phx
+
+    ; Init Input
+    call(Input_Init, player.input)
+    call(OAMManager_Init, player.oam_manager_ptr)
+    txa
+    clc
+    adc #player.oam_manager_ptr
+    tay
+
+    jsr Player_OAMRequest
+
+    plx
+    ply
+    pla
+    rts
+
+Player_OAMRequest:
+    pha
+    phy
+    call_ptr(OAMManager_Request, player.oam_manager_ptr) ; Request 1 OAM object
```

Comment:
This remains though. The logic would work if you follow the notes from above and game.asm.